Fix Google Font calls for different font weights

NatasaKralj opened the pull request:
This PR builds on top of https://github.com/mendix/docs/pull/4612, fixes https://github.com/mendix/docs/issues/4613. The call is built using [Google's tool ](https://fonts.google.com/noto/specimen/Noto+Sans?query=noto) to generate the API call for the following Noto Sans fonts:

- Regular 400 
- Regular 400 Italic
- Medium 500 
- Bold 700 

The site was using Regular 400 for every font-weight. Once the API calls are made, the fonts used on the page look different. This made it necessary to edit the scss files to match current font weight looks.

NatasaKralj committed:
Update Google Font calls for different font weights

MarkvanMents reviewed (commented):
Looks good. Would like to go through the font weight decisions - probably best to do that before merging rather than making a new PR.

NatasaKralj committed:
Update heading font weight and size

NatasaKralj committed:
Change position of 'draft' label

NatasaKralj committed:
Fix border indenting active ToC entry

NatasaKralj commented:
Updates:

- Removed SemiBold 600, no longer used anywhere
- Title and all headings switched to Regular 400
- Card title text made larger

Extra bits (not connected to fonts):

- Switched 'draft' label position to left side of link
- Adjusted ToC left border to always be present, just changes color on becoming active

MarkvanMents reviewed (approved):
Great. Thanks for working on this.
